tests/metrics/test_converters.py

```diff
@@ -114,14 +114,23 @@ def _setup_ddp(rank, worldsize):
     dist.init_process_group("gloo", rank=rank, world_size=worldsize)
 
 
-def _ddp_test_fn(rank, worldsize):
+def _ddp_test_fn(rank, worldsize, add_offset: bool, reduction_mean=False):
     _setup_ddp(rank, worldsize)
     tensor = torch.tensor([1.], device='cuda:0')
 
-    reduced_tensor = _sync_ddp_if_available(tensor)
+    if add_offset:
+        tensor = tensor + rank
+
+    if reduction_mean:
+        reduced_tensor = _sync_ddp_if_available(tensor, 'avg')
+
+        manual_reduction = sum([tensor.item() + i for i in range(dist.get_world_size())]) / dist.get_world_size()
+        assert reduced_tensor.item()
```

Comment:
I'm trying to understand this. How can this test fail? You are saying, if the reduction is not correct, `reduced_tensor.item() must be = 0`. Looking at the formula above, I don't see how that's possible.

Reply:
you're right, there is an issue, because it is supposed to be equal to `manual_reduction`